src/psyclone/psyGen.py

```diff
@@ -2909,11 +2927,30 @@ def coloured_text(self):
         '''
         return colored("KernCall", SCHEDULE_COLOUR_MAP["KernCall"])
 
-    def gen_code(self, parent):
+    def gen_code(self, parent, arguments=None):
+        '''
+        Generates the f2pygen AST of the Fortran for this kernel call and
+        writes the kernel itself to file if it has been transformed.
+
+        :param parent: The parent of this kernel call in the f2pygen AST.
+        :type parent: :py:calls:`psyclone.f2pygen.LoopGen`
+        :param arguments: The list of arguments to pass to the kernel.
+        :type arguments: list of str.
+        '''
         from psyclone.f2pygen import CallGen, UseGen
-        parent.add(CallGen(parent, self._name, self._arguments.arglist))
-        parent.add(UseGen(parent, name=self._module_name, only=True,
-                          funcnames=[self._name]))
+
+        # If kernel has been transformed then we rename it and write it to file
+        self.update()
```

Comment:
Could this have a name that better explains what it does e.g. rename_and_write(), or just write_to_file()?

Reply:
It could and now it has.